### PR TITLE
[Dashboard] Reorganize dashboard modules - actor

### PR DESCRIPTION
**Pull request by fyrestone**
## Why are these changes needed?

New actor module comes from,
- The logical_view module.
- Update actors from stats_collector module.

## Related issue number

https://github.com/ray-project/ray/issues/14766

## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Comment by rkooo567** [comment]
Hey @wuisawesome can you review if this doesn't impact any of changes you've made recently?

File: dashboard/datacenter.py [hunk 1]
```diff
@@ -1,7 +1,9 @@
 import logging
 import ray.new_dashboard.consts as dashboard_consts
 import ray.new_dashboard.memory_utils as memory_utils
-from ray.new_dashboard.actor_utils import actor_classname_from_task_spec
+# TODO(fyrestone): Not import from dashboard module.
```
Comment (rkooo567) [inline dashboard/datacenter.py:4]: What does this comment mean?

File: dashboard/modules/actor/actor_head.py [hunk 2]
```diff
@@ -0,0 +1,206 @@
...
+                    node_actors[actor_id] = actor_table_data
+                    DataSource.node_actors[node_id] = node_actors
+                # Update job actors.
+                job_actors = dict(DataSource.job_actors.get(job_id, {}))
+                job_actors[actor_id] = actor_table_data
+                DataSource.job_actors[job_id] = job_actors
+            except Exception:
+                logger.exception("Error receiving actor info.")
+
+    @routes.get("/logical/actor_groups")
```
Comment (rkooo567) [inline dashboard/modules/actor/actor_head.py:151]: Maybe renaming it to `all_actors`?

File: dashboard/modules/actor/actor_head.py [hunk 3]
```diff
@@ -0,0 +1,206 @@
...
+            try:
+                logger.info("Getting all actor info from GCS.")
+                request = gcs_service_pb2.GetAllActorInfoRequest()
+                reply = await self._gcs_actor_info_stub.GetAllActorInfo(
+                    request, timeout=5)
+                if reply.status.code == 0:
+                    actors = {}
+                    for message in reply.actor_table_data:
+                        actor_table_data = actor_table_data_to_dict(message)
+                        _process_actor_table_data(actor_table_data)
```
Comment (rkooo567) [inline dashboard/modules/actor/actor_head.py:85]: Just removing this function? It is just two lines.

**Review by rkooo567 — APPROVED** [review]
LGTM. It is mostly refactoring right?

File: dashboard/datacenter.py (hunk 1 above)
Comment (fyrestone) [reply dashboard/datacenter.py:4]: The dashboard is best not to rely on a module. But, we can treat the module as required.

File: dashboard/modules/actor/actor_head.py (hunk 2 above)
Comment (fyrestone) [reply dashboard/modules/actor/actor_head.py:151]: This API is used by the frontend, I just moved it from the logical_view module. Do we really need to rename it to `all_actors`?

**Review by WangTaoTheTonic — APPROVED** [review]
It's all code reconstruct. I'm gonna click approve.

**Review by wuisawesome — APPROVED** [review]
Lgtm

**Pull request merged**